x/emissions/keeper/msgserver/msg_server_requests.go

```diff
@@ -3,143 +3,87 @@ package msgserver
-			return nil, err
-		}
-		if request.Cadence < minFastestAllowedCadence {
-			return nil, types.ErrInferenceRequestCadenceTooFast
-		}
-		// 7. Check the cadence is no more than the maximum allowed slowest cadence
-		maxSlowestAllowedCadence, err := ms.k.GetParamsMaxRequestCadence(ctx)
-		if err != nil {
-			return nil, err
-		}
-		if request.Cadence > maxSlowestAllowedCadence {
-			return nil, types.ErrInferenceRequestCadenceTooSlow
-		}
-	}
-	// 8. Check the cadence is not greater than the block valid until
-	if currentBlock+request.Cadence > request.BlockValidUntil {
-		return nil, types.ErrInferenceRequestWillNeverBeScheduled
-	}
-	MinRequestUnmetDemand, err := ms.k.GetParamsMinRequestUnmetDemand(ctx)
+
+	// Check that the request isn't spam by checking that the amount of funds it bids is greater than a global minimum demand per request
+	epsilon, err := ms.k.GetParamsEpsilon(ctx)
 	if err != nil {
 		return nil, err
 	}
-	// Check that the request isn't spam by checking that the amount of funds it bids is greater than a global minimum demand per request
-	if request.BidAmount.LT(MinRequestUnmetDemand) {
-		return nil, types.ErrInferenceRequestBidAmountTooLow
+	amountDec := alloraMath.NewDecFromInt64(msg.Amount.Int64())
+	if amountDec.Lte(epsilon) {
+		return nil, types.ErrFundAmountTooLow
 	}
-	// 9. Check sender has funds to pay for the inference request
+	// Check sender has funds to pay for the inference request
 	// bank module does this for us in module SendCoins / subUnlockedCoins so we don't need to check
-	// 10. Send funds
-	senderAddr, err := sdk.AccAddressFromBech32(request.Sender)
+	// Send funds
+	senderAddr, err := sdk.AccAddressFromBech32(msg.Sender)
 	if err != nil {
 		return nil, err
 	}
-	amountInt := cosmosMath.NewIntFromBigInt(request.BidAmount.BigInt())
-	coins := sdk.NewCoins(sdk.NewCoin(params.DefaultBondDenom, amountInt))
+	amountInt := cosmosMath.NewIntFromBigInt(msg.Amount.BigInt())
+	coins := sdk.NewCoins(sdk.NewCoin(appParams.DefaultBondDenom, amountInt))
 	err = ms.k.SendCoinsFromAccountToModule(ctx, senderAddr, types.AlloraRequestsAccountName, coins)
 	if err != nil {
 		return nil, err
 	}
-	// 11. record the number of tokens sent to the module account
-	err = ms.k.SetRequestDemand(ctx, requestId, request.BidAmount)
-	if err != nil {
-		return nil, err
-	}
-	// 12. Write request state into the mempool state
-	request.BlockLastChecked = currentBlock
-	unmetDemand, err := ms.k.AddToMempool(ctx, *request)
+
+	// Account for the revenue the topic has generated
+	err = ms.k.AddTopicFeeRevenue(ctx, msg.TopicId, msg.Amount)
 	if err != nil {
 		return nil, err
 	}
-	// 13. Activate topic if meet demand
-	isActivated, err := ms.k.IsTopicActive(ctx, request.TopicId)
+
+	// Activate topic if it exhibits minimum weight
+	isActivated, err := ms.k.IsTopicActive(ctx, msg.TopicId)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "error getting topic activation status")
 	}
 	if !isActivated {
```

Comment:
this if !isActivated branch should probably be put into its own function separately and shared with an ActivateTopic call